libpod/networking_linux.go

```diff
@@ -1065,11 +1088,71 @@ func (c *Container) getContainerNetworkInfo() (*define.InspectNetworkSettings, e
 	return settings, nil
 }
 
+func (c *Container) inspectJoinedNetworkNS(networkns string) (q types.StatusBlock, retErr error) {
+	var result types.StatusBlock
+	err := ns.WithNetNSPath(networkns, func(_ ns.NetNS) error {
+		ifaces, err := net.Interfaces()
+		if err != nil {
+			return err
+		}
+		routes, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+		if err != nil {
+			return err
+		}
+		var gateway net.IP
+		for _, route := range routes {
+			// default gateway
+			if route.Dst == nil {
+				gateway = route.Gw
+			}
+		}
+		result.Interfaces = make(map[string]types.NetInterface)
+		for _, iface := range ifaces {
+			if strings.Contains(iface.Flags.String(), "loopback") {
+				continue
+			}
+			addrs, err := iface.Addrs()
+			if err != nil {
+				continue
+			}
+			if len(addrs) == 0 {
+				continue
+			}
+			subnets := make([]types.NetAddress, 0)
+			for _, address := range addrs {
+				if ipnet, ok := address.(*net.IPNet); ok {
+					subnets = append(subnets, types.NetAddress{
+						IPNet: types.IPNet{
+							IPNet: *ipnet,
+						},
+						Gateway: gateway,
+					})
+				}
+			}
+			if macAddress, err := net.ParseMAC(iface.HardwareAddr.String()); err == nil {
```

Comment:
Why do you pare the mac address again, just use `iface.HardwareAddr` below instead of `macAddress`

Reply:
They are two different types
- net.HardwareAddr
- "github.com/containers/common/libnetwork/types".HardwareAddr
and I am neither a go expert nor networking, and that was my gullible solution.

Reply:
but macAddress is also net.HardwareAddr
it should work directly with `types.HardwareAddr(iface.HardwareAddr)` instead of parsing it again

Reply:
Absolutely right, thanks.